devlib/trace/perf.py

```diff
@@ -131,7 +163,25 @@ def _get_target_outfile(self, label):
     def _build_perf_command(self, options, events, label):
         event_string = ' '.join(['-e {}'.format(e) for e in events])
         command = PERF_COMMAND_TEMPLATE.format(self.binary,
+                                               self.perf_mode,
                                                options or '',
                                                event_string,
                                                self._get_target_outfile(label))
         return command
+
+    def _get_target_perfdatafile(self):
+        return self.target.get_workpath('perf.data')
+
+    def _get_target_reportfile(self, label):
+        return self.target.get_workpath('{}.rpt'.format(label))
+
+    def _get_target_tracefile(self, label):
+        return self.target.get_workpath('{}.trace'.format(label))
+
+    def _build_perf_report_command(self, reportoptions, label):
+        reportoptions_string = reportoptions
+        command = PERF_REPORT_COMMAND_TEMPLATE.format(self.binary,
+                                                      reportoptions_string,
+                                                      self._get_target_perfdatafile(),
+                                                      self._get_target_reportfile(label))
```

Comment:
And when testing this on my android device the fails with error code 244 caused by `magic/endian check failed`. Have you seen this on your end?

Reply:
I have seen error code 244 on a few occasions but haven't tracked it down. It has been working reliable for a while for me on the Android device I am using. I was guessing it was related to the `perf report` command being '<command>' | su rather than '<command> | su' used for `perf record` but that just a guess. I'll try it on a few devices to see if it reliable reproduces or not.

Reply:
I've tested on a few Android devices and see error code 244 or just failures on some but not others. It seems to be consistent behavior on the devices, so a device either works or doesn't.

The error message seems to be related to some check of the perf.data file and likely finding an corruption issue with perf.data. So I think this is a result of the approach WA uses for running (aka killing) perf and some platforms behavior writing to the perf.data file to storage when killed. 

Apart from a re-write of perf support in WA I think the only thing that could be done here is to try identify the behavior that causes the file corruption and see if we can override it from the device command prompt (e.g. setprop ....)

Reply:
That's interesting, I've tried running this again with the latest version and no longer see this problem on my device. 
And I apologies however I think I may have made an error before. In order to stop the perf instrument we're actually just killing the `sleep` command that perf is monitoring rather than killing `perf` itself in order to stop the measurements. The killing of `perf` only happens when the instrument is reset to ensure there are no extra processes running before running the workload and in the teardown stage. So I'm not sure this would explain this behaviour. 
I'm instead wondering if it could be down to the speed of the device and if the data hasn't finished being written to disk before we try to process it? Would it be possible to try adding a delay in between the two and then test this on a device that has this issue to see if that could be the case?

Reply:
I hadn't notice @marcbonnici had replied to this: could we try rebasing on #384 to see if the error message still appears?

Reply:
I've tried #384  and sleep(10) and still seeing exit code 244.

Reply:
Hmm... I'm wondering why I received this initially and haven't been able to reproduce it since.  
Would you be able to run the steps that WA is performing manually to determine if it is something that WA is doing to cause this or whether it's perf on the device itself that is having issues?

Reply:
Hmm... I'm wondering why I received this initially and haven't been able to reproduce it since.
Would you be able to run the steps that WA is performing manually to determine if it is something that WA is doing to cause this or whether it's perf on the device itself that is having issues?

Reply:
Just in case, could you give #386 a try?